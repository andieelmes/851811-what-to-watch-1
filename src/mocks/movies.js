export const MOVIES = [
  {
    id: `fantastic-beasts-the-crimes-of-grindelwald`,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
  },
  {
    id: `bohemian-rhapsody`,
    img: `img/bohemian-rhapsody.jpg`,
    title: `Bohemian Rhapsody`,
  },
  {
    id: `seven-years-in-tibet`,
    img: `img/seven-years-in-tibet.jpg`,
    title: `Seven Years in Tibet`,
  },
  {
    id: `what-we-do-in-the-shadows`,
    img: `img/what-we-do-in-the-shadows.jpg`,
    title: `What We Do in the Shadows`,
  },
  {
    id: `pulp-fiction`,
    img: `img/pulp-fiction.jpg`,
    title: `Pulp Fiction`,
  },
];
