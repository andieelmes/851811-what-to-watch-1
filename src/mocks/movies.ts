export const MOVIES = [
  {
    id: 1,
    img: 'https://es31-server.appspot.com/wtw/static/film/background/Midnight_Special.jpg',
    title: 'Midnight Special',
    preview: 'https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4',
    video: 'http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4',
    genre: 'action',
    poster: 'https://es31-server.appspot.com/wtw/static/film/poster/Midnight_Special.jpg',
    backgroundImage: 'https://es31-server.appspot.com/wtw/static/film/background/Midnight_Special.jpg',
    backgroundColor: '#828585',
    description: 'A father and son go on the run, pursued by the government and a cult drawn to the child\'s special powers.',
    rating: 3.3,
    ratingsCount: 67815,
    director: 'Jeff Nichols',
    starring: [
      'Michael Shannon',
      'Joel Edgerton',
      'Kirsten Dunst '
    ],
    duration: 112,
    year: 2016,
    favorite: false
  },
  {
    id: 2,
    img: 'https://es31-server.appspot.com/wtw/static/film/background/What-We-Do-in-the-Shadows.jpg',
    title: 'What We Do in the Shadows',
    preview: 'https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm',
    video: 'http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4',
    genre: 'comedy',
    poster: 'https://es31-server.appspot.com/wtw/static/film/poster/What-We-Do-in-the-Shadows.jpg',
    backgroundImage: 'https://es31-server.appspot.com/wtw/static/film/background/What-We-Do-in-the-Shadows.jpg',
    backgroundColor: '#A39E81',
    description: 'A look into the daily (or rather, nightly) lives of three vampires who\'ve lived together for over 100 years, in Staten Island.',
    rating: 4.2,
    ratingsCount: 6173,
    director: 'Jemaine Clement',
    starring: [
      'Kayvan Novak',
      'Matt Berry',
      'Natasia Demetriou'
    ],
    duration: 30,
    year: 2019,
    favorite: true
  }
];

export const GENRES = [
  `all genres`, `comedies`, `crime`, `documentaries`, `dramas`,
];
