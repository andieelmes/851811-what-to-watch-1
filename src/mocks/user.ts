export const USER = {
  avatar: '/',
  authorized: true,
  name: 'name'
}
