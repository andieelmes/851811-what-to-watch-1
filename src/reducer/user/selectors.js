import Namespace from "../namespaces";

const NAMESPACE = Namespace.USER;


export const getAuthorizationStatus = (state) => {
  return state[NAMESPACE].isAuthorizationRequired;
};
