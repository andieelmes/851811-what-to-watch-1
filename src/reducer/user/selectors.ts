import Namespace from "../namespaces";

const NAMESPACE = Namespace.USER;


export const getAuthorizationStatus = (state) => {
  return !state[NAMESPACE].isAuthorizationRequired;
};

export const getUserInfo = (state) => {
  const {
    isAuthorizationRequired,
    ...userInfo
  } = state[NAMESPACE];
  return userInfo;
};
