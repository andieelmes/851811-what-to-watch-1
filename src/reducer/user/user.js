import {ENDPOINT_URL} from 'server-variables';

const initialState = {
  isAuthorizationRequired: true,
  avatar: ``,
  name: ``,
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  GET_USER_DATA: `GET_USER_DATA`
};

const ActionCreator = {
  requireAuthorization: (status) => {
    return {
      type: ActionType.REQUIRED_AUTHORIZATION,
      payload: status,
    };
  },

  getUserData: (data) => {
    return {
      type: ActionType.GET_USER_DATA,
      payload: data,
    };
  },
};

const Operation = {
  login: (email, password) => (dispatch, _getState, api) => {
    return api.post(`/login`, {email, password})
      .then((response) => {
        const data = {
          avatar: `${ENDPOINT_URL}${response.data.avatar_url.replace(`/wtw`, ``)}`,
          name: response.data.name,
        }
        dispatch(ActionCreator.requireAuthorization(false));
        dispatch(ActionCreator.getUserData(data));
      })
      .catch((error) => {
        console.log(error);
      })
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return {...state, isAuthorizationRequired: action.payload}
    case ActionType.GET_USER_DATA: return {...state, ...action.payload}
  }

  return state;
};


export {
  ActionCreator,
  ActionType,
  Operation,
  reducer,
};
