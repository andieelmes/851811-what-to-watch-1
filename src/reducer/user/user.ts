import {ENDPOINT_URL} from 'App/server-variables';
import { Action, ActionCreator, AnyAction } from 'redux';
import { ThunkDispatch, ThunkAction } from 'redux-thunk';

import {User} from 'App/types';
import { Dispatch } from 'react';

const initialState: User = {
  isAuthorizationRequired: true,
  avatar: ``,
  name: ``,
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  GET_USER_DATA: `GET_USER_DATA`
};

const requireAuthorization: ActionCreator<Action> = (status: boolean) => {
  return {
    type: ActionType.REQUIRED_AUTHORIZATION,
    payload: status,
  };
};

const getUserData: ActionCreator<Action> = (data: {avatar: string, name: string}) => {
  return {
    type: ActionType.GET_USER_DATA,
    payload: data,
  };
};

const ActionCreator = {
  requireAuthorization: requireAuthorization,
  getUserData: getUserData,
};

const onLoginSuccess = (response, dispatch: Dispatch<any>) => {
  const data = {
    avatar: `${ENDPOINT_URL}${response.data.avatar_url.replace(`/wtw`, ``)}`,
    name: response.data.name,
  }
  dispatch(ActionCreator.requireAuthorization(false));
  dispatch(ActionCreator.getUserData(data));
}

const Operation = {
  postLogin: (email: string, password: string, callback: () => void): ThunkAction<void, User, null, AnyAction> => (dispatch: ThunkDispatch<User, void, AnyAction>, _getState: () => User, api: any) => {
    return api.post(`/login`, {email, password})
      .then((response) =>
        {
          onLoginSuccess(response, dispatch);
          callback();
        }
      )
      .catch((error) => {
        console.log(error);
      })
  },
  getLogin: () :ThunkAction<void, User, null, AnyAction> => (dispatch: ThunkDispatch<User, void, AnyAction>, _getState: () => User, api: any) => {
    return api.get(`/login`)
      .then((response) => onLoginSuccess(response, dispatch))
      .catch((error) => {
        console.log(error);
      })
  },
}

const reducer = (state = initialState, action: AnyAction) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return {...state, isAuthorizationRequired: action.payload}
    case ActionType.GET_USER_DATA: return {...state, ...action.payload}
  }

  return state;
};


export {
  ActionCreator,
  ActionType,
  Operation,
  reducer,
};
