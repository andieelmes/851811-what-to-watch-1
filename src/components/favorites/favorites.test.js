import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import Favorites from 'App/components/favorites/favorites';

import {USER} from 'App/mocks/user';
import {MOVIES} from 'App/mocks/movies';

describe(`Favorites component`, () => {
  it(`should render correctly`, () => {
    const tree = renderer
      .create(<BrowserRouter><Favorites user={USER} movies={MOVIES}/></BrowserRouter>, {createNodeMock: (el) => el})
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
