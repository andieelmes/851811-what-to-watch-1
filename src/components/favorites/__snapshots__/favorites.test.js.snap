// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites component should render correctly 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <img
          alt="User avatar of name"
          height="63"
          src="/"
          width="63"
        />
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        data-movie-id={1}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__image"
          href="/film/1"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
        >
          <img
            alt="Midnight Special"
            height="175"
            src="https://es31-server.appspot.com/wtw/static/film/background/Midnight_Special.jpg"
            style={null}
            width="280"
          />
          <video
            controls={true}
            loop={true}
            muted={true}
            poster="https://es31-server.appspot.com/wtw/static/film/background/Midnight_Special.jpg"
            style={
              Object {
                "display": "none",
              }
            }
          />
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/1"
            onClick={[Function]}
          >
            Midnight Special
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        data-movie-id={2}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <a
          className="small-movie-card__image"
          href="/film/2"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
        >
          <img
            alt="What We Do in the Shadows"
            height="175"
            src="https://es31-server.appspot.com/wtw/static/film/background/What-We-Do-in-the-Shadows.jpg"
            style={null}
            width="280"
          />
          <video
            controls={true}
            loop={true}
            muted={true}
            poster="https://es31-server.appspot.com/wtw/static/film/background/What-We-Do-in-the-Shadows.jpg"
            style={
              Object {
                "display": "none",
              }
            }
          />
        </a>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/film/2"
            onClick={[Function]}
          >
            What We Do in the Shadows
          </a>
        </h3>
      </article>
    </div>
  </section>
  <footer
    className="page-footer"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link logo__link--light"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="copyright"
    >
      <p>
        Â© 2019 What to watch Ltd.
      </p>
    </div>
  </footer>
</div>
`;
